lib/cursor.js

```diff
@@ -171,6 +171,27 @@ Cursor.prototype.cursorSkip = function() {
   return this.cursorState.skip;
 };
 
+Cursor.prototype._endSession = function(options, callback) {
```

Comment:
this is more of a nitpick/question than a mandate, but for private methods like this I generally keep them _off_ the prototype unless I explicitly need it for testing. I would opt for a:
```
function cursorEndSession(cursor, options, callback)
```
that isn't exported by the file.

Is there a reason you prefer your approach here? Do you anticipate people will need to override this behavior, or that we will need to futz with it in testing?

Reply:
The method is overridden in the native PR [here](https://github.com/mongodb/node-mongodb-native/pull/1668/files#diff-14b597bc6604e1b7ec69ae786619e979R217). I'm not particularly happy about this though.